src/parser/grammar.mly

```diff
@@ -165,16 +167,23 @@ chr_rule:
 
 pred:
 | attributes = attributes; PRED;
-  c = constant; args = separated_list(option(CONJ),pred_item) {
-   let name = c in
-   { Mode.loc=loc $sloc; name; args = List.map fst args },
-   { Type.loc=loc $sloc; attributes; name;
-     ty = List.fold_right (fun (_,t) ty ->
-       mkApp (loc $loc(c)) [mkCon "->";t;ty]) args (mkCon "prop") }
+  name = constant; args = separated_list(option(CONJ),pred_item) {
+   { Type.loc=loc $sloc; name; attributes; ty = TPred ([], args @ [mode_of_IO 'o', TConst (Func.from_string "prop")]) }
  }
 pred_item:
 | io = IO_COLON; ty = type_term { (mode_of_IO io,ty) }
 
+anonymous_pred:
```

Comment:
This rule is kept in order to check that modes are no more used, in which case an error is raised.
Maybe there is a more elegant way of throwing the error without passing though this rule: I think for example to a error given by the lexer